src/librustc_mir/transform/simplify_try.rs

```diff
@@ -418,9 +422,15 @@ impl<'tcx> MirPass<'tcx> for SimplifyArmIdentity {
                 // Fix the debug info to point to the right local
                 for dbg_index in opt_info.dbg_info_to_adjust {
                     let dbg_info = &mut debug_info[dbg_index];
-                    assert!(dbg_info.place.projection.is_empty());
-                    dbg_info.place.local = opt_info.local_0;
-                    dbg_info.place.projection = opt_info.dbg_projection;
+                    assert!(
+                        matches!(dbg_info.value, VarDebugInfoContents::Place(_)),
+                        "value was not a Place"
+                    );
+                    if let VarDebugInfoContents::Place(mut p) = dbg_info.value {
```

Comment:
Ouch, this doesn't do what you think. Since `VarDebugInfoContents` is `Copy`, this creates a new copy and mutates it. This should be 

```suggestion
                    if let VarDebugInfoContents::Place(p) = &mut dbg_info.value {
```